nautobot/wireless/models.py

```diff
@@ -0,0 +1,257 @@
+        related_name="radio_profiles",
+        blank=True,
+    )
+    regulatory_domain = models.CharField(max_length=CHARFIELD_MAX_LENGTH)
+    rx_power_min = models.IntegerField()
+
+    class Meta:
+        ordering = ["name"]
+
+    def __str__(self):
+        return self.name
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "webhooks",
+)
+class WirelessNetwork(PrimaryModel):
+    """
+    A WirelessNetwork represents a wireless network that can be broadcast by an AccessPoint.
+    """
+
+    name = models.CharField(max_length=CHARFIELD_MAX_LENGTH, unique=True)
+    description = models.CharField(max_length=200, blank=True)
+    ssid = models.CharField(max_length=CHARFIELD_MAX_LENGTH)
+    mode = models.CharField(
+        max_length=CHARFIELD_MAX_LENGTH,
+        choices=WirelessDeploymentModeChoices,
+        default=WirelessDeploymentModeChoices.CENTRAL,
+    )
+    enabled = models.BooleanField(default=True)
+    authentication = models.CharField(
+        max_length=CHARFIELD_MAX_LENGTH,
+        choices=WirelessAuthTypeChoices,
+        default=WirelessAuthTypeChoices.OPEN,
+    )
+    secrets_group = models.ForeignKey(
+        to="extras.SecretsGroup",
+        on_delete=models.SET_NULL,
+        default=None,
+        blank=True,
+        null=True,
+    )
+    hidden = models.BooleanField(default=False)
+    vlans = models.ManyToManyField(
+        to="ipam.VLAN",
+        related_name="wireless_networks",
+        blank=True,
+    )
+
+    class Meta:
+        ordering = ["name"]
+
+    def __str__(self):
+        return self.name
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+)
+class AccessPointGroupWirelessNetworkAssignment(BaseModel):
+    """
+    An AccessPointGroupWirelessNetworkAssignment represents the assignment of a WirelessNetwork to an AccessPointGroup.
+    """
+
+    access_point_group = models.ForeignKey(
+        to="wireless.AccessPointGroup",
+        on_delete=models.CASCADE,
+        related_name="wireless_network_assignments",
+    )
+    wireless_network = models.ForeignKey(
+        to="wireless.WirelessNetwork",
+        on_delete=models.CASCADE,
+        related_name="access_point_group_assignments",
+    )
+    vlan = models.ForeignKey(
+        to="ipam.VLAN",
+        on_delete=models.SET_NULL,
+        related_name="access_point_group_wireless_network_assignments",
+        blank=True,
+        null=True,
+    )
+    is_metadata_associable_model = False
+
+    class Meta:
+        unique_together = ["access_point_group", "wireless_network"]
+        ordering = ["access_point_group", "wireless_network"]
+
+    def __str__(self):
+        return f"{self.access_point_group}: {self.wireless_network}"
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+)
+class AccessPointGroupRadioProfile(BaseModel):
```

Comment:
Are we creating explicit through tables for this m2m and the one below because we plan to add more fields in the future?

Reply:
Any M2M that needs to be manageable through the REST API needs an explicit through table.

Reply:
I was under the impression that we wanted explicit through tables going forward.